Add authentication and split into signup also

mattjbrent opened the pull request:
* split up authentication into signup as it follows the pattern a little better
* Add pngs of finished wireframes

mattjbrent committed:
Add authentication and split into signup also

* split up authentication into signup as it follows the pattern a little better
* Add pngs of finished wireframes

cellio commented on scoped/authentication/authentication-indirect.md:
```diff
@@ -0,0 +1,38 @@
+----
+**Basic flow**
+
+- User sees "sign in to answer this question" link/button and clicks.
```
I guess for consistency we should say "log in or sign up", not "sign in" which is probably my typo.

cellio commented on scoped/sign-up/sign-up.md:
```diff
@@ -0,0 +1,42 @@
+- User presented with short form and explanation summary
+- User briefly reads paragraph, enters display name, email and password
+- Any appropriate prompts and error messages are displayed underneath the fields
+- User clicks sign up, once successful user is redirected to login page
```
We need to account for the "has account on another site on the network, wants to add this site" case.  That can be a separate use case (linked to this one); let's not lose track of it.

cellio commented on scoped/authentication/login-indirect/login-indirect.md:
```diff
@@ -0,0 +1,32 @@
+- User presented with various prompts under the fields based on action
+- Any errors are surfaced in the same manner, errors are clearly identified using red
+- Form will not submit unless all fields are filled
+- Form cannot resubmit if user has already submitted form
```
Can't comment on scoped/authentication/login-indirect/login-indirect.png so commenting here.  Is "want to answer?" in the diagram contextual?  (e.g. would it be "want to post?" or "want to vote?" etc for other actions?)  We don't want to show "want to answer?" if the person tried to ask a question; either using generic language reflecting the specific action that was attempted is fine.

mattjbrent replied:
This would be contextual to the site actually. Answer could be anything. Perhaps wrapping in `${}` might denote this better?

cellio replied:
I'm not sure what you mean by "to the site"; it seems more contextual to the *action*, like what button/link the person clicked on before getting told to log in.  On a question page this would be "log in to answer", which was the use case that led to this I think, but the same issue would arise of someone who's not logged in clicks on "ask question".  So either that top line needs to vary based on how the user came to be facing that dialogue, or we need to do something generic.

mattjbrent replied:
sorry im referring to the instance - so writing.codidact.com vs programming.codidact.com might use different terminology or is there a way for us to just use a more generic term here like post in your example

cellio replied:
Let's just do that -- instead of "answer" say "post", and that will cover us for a while.  We can revisit when we have other use cases.  I'm not trying to make this complicated. :-)

mattjbrent committed:
Fix log in sign up terminology for authentication-indirect

Fix sign up flow by adding another network use case
r with '#' will be ignored, and an empty message aborts the commit.

mattjbrent committed:
Update login-indirect.png, changing question to post
